it/v1/endpoints/IgnoreEmploymentExpensesControllerISpec.scala

```diff
@@ -0,0 +1,206 @@
+
+  private trait Test {
+
+    val nino: String = "AA123456A"
+    val taxYear: String = "2019-20"
+
+    val requestBody: JsValue = Json.parse(
+      s"""
+         |{
+         |  "ignoreExpenses": true
+         |}
+         |""".stripMargin
+    )
+
+    val responseBody: JsValue = Json.parse(
+      s"""
+         |{
+         |  "links": [
+         |    {
+         |      "href": "/individuals/expenses/employments/$nino/$taxYear",
+         |      "method": "GET",
+         |      "rel": "self"
+         |    },
+         |    {
+         |      "href": "/individuals/expenses/employments/$nino/$taxYear/ignore",
+         |      "method": "PUT",
+         |      "rel": "ignore-employment-expenses"
+         |    }
+         |  ]
+         |}
+         |""".stripMargin)
+
+    def uri: String = s"/employments/$nino/$taxYear/ignore"
+
+    def desUri: String = s"/income-tax/expenses/employments/$nino/$taxYear"
+
+    def setupStubs(): StubMapping
+
+    def request(): WSRequest = {
+      setupStubs()
+      buildRequest(uri)
+        .withHttpHeaders((ACCEPT, "application/vnd.hmrc.1.0+json"))
+    }
+
+    def errorBody(code: String): String =
+      s"""
+         |      {
+         |        "code": "$code",
+         |        "reason": "des message"
+         |      }
+    """.stripMargin
+  }
+
+  "Calling the ignore endpoint" should {
+
+    "return a 200 status code" when {
+
+      "any valid request is made" in new Test {
+
+        override def setupStubs(): StubMapping = {
+          AuditStub.audit()
+          AuthStub.authorised()
+          MtdIdLookupStub.ninoFound(nino)
+          DesStub.onSuccess(DesStub.PUT, desUri, NO_CONTENT, JsObject.empty)
+        }
+
+        val response: WSResponse = await(request().put(requestBody))
+        response.status shouldBe OK
+        response.json shouldBe responseBody
+        response.header("X-CorrelationId").nonEmpty shouldBe true
+      }
+    }
+
+    "return error according to spec" when {
+
+      "validation error" when {
+        s"an invalid NINO is provided" in new Test {
+          override val nino: String = "INVALID_NINO"
+
+          override def setupStubs(): StubMapping = {
+            AuditStub.audit()
+            AuthStub.authorised()
+            MtdIdLookupStub.ninoFound(nino)
+          }
+
+          val response: WSResponse = await(request().put(requestBody))
+          response.status shouldBe BAD_REQUEST
+          response.json shouldBe Json.toJson(NinoFormatError)
+        }
+        s"an invalid taxYear is provided" in new Test {
+          override val taxYear: String = "INVALID_TAXYEAR"
+
+          override def setupStubs(): StubMapping = {
+            AuditStub.audit()
+            AuthStub.authorised()
+            MtdIdLookupStub.ninoFound(nino)
+          }
+
+          val response: WSResponse = await(request().put(requestBody))
+          response.status shouldBe BAD_REQUEST
+          response.json shouldBe Json.toJson(TaxYearFormatError)
```

Comment:
It seems to me at least the ISpec is testing for the FORMAT_NINO error and FORMAT_TAX_YEAR error twice, one individually and one in the Seq